lib/inspector/promises.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const inspector = require('inspector');
+const { promisify } = require('util');
+class Session extends inspector.Session {
+
+  /**
+   * Posts a message to the inspector back-end.
+   * @param {string} method
+   * @param {Record<unknown, unknown>} [params]
+   * @returns {Promise}
+   */
+  async post(method, params) {
+    return promisify(super.post.bind(this))(method, params);
```

Comment:
```suggestion
  #post = promisify(super.post.bind(this));

  /**
   * Posts a message to the inspector back-end.
   * @param {string} method
   * @param {Record<unknown, unknown>} [params]
   * @returns {Promise}
   */
  async post(method, params) {
    return this.#post(method, params);
```
this way, the original method is looked up once, the bound method is created once, and cached privately on the instance.

Reply:
uhuu nice! I loved it! Thanks